src/economy/economy.cpp

```diff
@@ -396,6 +443,7 @@ void adjust_artisan_balance(sys::state& state, dcon::nation_id n) {
 
 				total_ex_weights += w;
 				temp_weights[cid.index()] = profit * profit;
+				profits[cid.index()] = profit;
```

Comment:
you don't need to store both profits and temp weights. Just store the profit and then square it as needed when the temp weight was previously called for